### ISSUE TITLE
`CA1000` hasn't been reconsidered with static abstracts in mind

### ISSUE THREAD

**Issue by RenderMichael** [dotnet/roslyn-analyzers#7126]
### Analyzer

**Diagnostic ID**: [CA1000](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1000): `Do not declare static members on generic types`

### Analyzer source

.NET 8.0.100 SDK

The pattern adopted for generic math in .NET 7, known as "Curiously recurring template pattern", violates this rule. Of course a rule like this one isn't enabled in the runtime repo, but when I tried to do the same I got this diagnostic pop up.

### Steps To Reproduce

Using .NET >=7:
``` csharp
public interface IConfigurationFileProvider<TConfiguration>
{
    public static abstract TConfiguration FromFile(string path);
}
```

### Expected behavior

When the member is abstract, this diagnostic should not fire.

### Actual behavior

This diagnostic fires.

## Additional Context

The above repro is a minimal repro, but this diagnostic still fires when using the recursive pattern:

``` csharp
public interface IConfigurationFileProvider<TConfiguration>
    where TConfiguration : IConfigurationFileProvider<TConfiguration>
{
    public static abstract TConfiguration FromFile(string path);
}
```

**Comment by mavasani** [comment]
Sounds reasonable to me. @buyaa-n?

**Comment by buyaa-n** [comment]
Sounds good to me too

**Comment by RenderMichael** [comment]
Can this bug be assigned to me? It looks like a straightforward fix.

**Comment by RenderMichael** [comment]
Opened PR #7142
